microsoft/azure-databricks-operator
# Contribute load testing and mock api

This is a PR to contribute back the load testing framework built by our dev crew (@stuartleeks @lawrencegripper @storey247 @martinpeck @damoodamoo @bart-jansen) at Microsoft.

PR contains:
- Locust load testing framework + load test scenarios for the Databricks Operator
- Databrick Mock API used to run the load tests against
- Install python and initalize virutal enviroment in dev container
- Load testing and MockAPI dashboards in grafana
- Make commands to:
   - test and run Locust, mockAP
   - run load tests in a kind cluster
- Docs for MockAPI and Locust usage
- Go script to verify load tests have run successfully
- Load tests added to the build pipeline

Possible future improvements:
- Use the mockAPI for part of the integration tests
- Avoid spinning up the kind cluster twice for integration tests and load tests
- Display load test result charts in the PR after completion
